diagnostic.md

```diff
@@ -6,70 +6,92 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What are two ways of telling rails where to route a GET request to '/books/4'?
 
 ```rb
-// your response here
+Rails.application.routes.draw do
+  resources :books, only: [:show]
+
+  post '/sign-up' => 'users#signup'
+  post '/sign-in' => 'users#signin'
+  delete '/sign-out/:id' => 'users#signout'
+  patch '/change-password/:id' => 'users#changepw'
+  resources :users, only: [:index, :show]
+end
+
+or
+
+Rails.application.routes.draw do
+  resources :books, except: [:new, :edit, :create, :destroy, :update, :index]
+
+  post '/sign-up' => 'users#signup'
+  post '/sign-in' => 'users#signin'
+  delete '/sign-out/:id' => 'users#signout'
+  patch '/change-password/:id' => 'users#changepw'
+  resources :users, only: [:index, :show]
+end
+
+But that latter one is kind of excessive.
 ```
 
 In what file does the above code go?
 
 ```md
-// your response here
+config/routes.rb
 ```
 
 In your own words, what does the command `bundle exec rake db:migrate` do?
 
 ```md
-// your response here
+It creates a migration for your database with correct labelling (e.g. CreateBooks instead of CreateBook), and the migration lets you make changes to your database.
 ```
 
 What is the difference between `rails g scaffold things` and
 `rails g controller things`?
 
 ```md
-// your response here
+I believe using `rails g scaffold things` generates a serializer file, a model file, and a controller file, whereas `rails g controller things` will just generate a controller file.
 ```
 
 What is the job of the `serializer`?
 
 ```md
-// your response here
+The serializer is used to send data back to the client... it works kind of like a filter for what the server can send out or receive.
```

Comment:
Serializers filter and reshape data, but don't actually send the data back to the client.